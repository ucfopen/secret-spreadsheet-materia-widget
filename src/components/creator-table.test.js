import React from 'react'
import Table from './creator-table'
import renderer from 'react-test-renderer'
import { mount } from '../enzyme'
import document from './creator-table'


const cellData = (value, check) => {
	return {
		'materiaType': 'question',
		'id': null,
		'type': 'QA',
		'options': {
			'blank': check,
		},
		'questions': [{
			'text': value,
		}],
		'answers': [{
			'id': null,
			'text': value,
			'value': 100
		}]
	}
}

const makeQset = (text = '', blank = false, left = true, header = true) => {
	return {
		dimensions: {
			x: 1,
			y: 1,
		},
		items: [{
			items: [[
				cellData(text, blank),
			]]
		}],
		left: left,
		header: header,
	}
}

describe('CreatorTable component', function() {

	beforeEach(() => {
		jest.resetModules()

	})

	test('CreatorTable renders 1x1 table, left aligned, header', () => {
		const qset = makeQset()

		const props = {
			qset,
			cellData: jest.fn(),
		}
		const component = mount(<Table {... props}/>)

		const tree = renderer.create(<Table {... props}/>).toJSON()
		expect(tree).toMatchSnapshot()

	})

	test('CreatorTable renders 1x1 table, center aligned, no header', () => {
		const qset = makeQset('', false, false, false)

		const props = {
			qset,
			cellData: jest.fn(),
		}
		const component = mount(<Table {... props}/>)

		const tree = renderer.create(<Table {... props}/>).toJSON()
		expect(tree).toMatchSnapshot()

	})

	test('CreatorTable renders 1x1 table, unable to remove 1st row', () => {
		const qset = makeQset()

		const props = {
			qset,
			cellData: jest.fn(),
		}
		const component = mount(<Table {... props}/>)

		expect(props.qset.dimensions.x).toEqual(1)
		component.instance().removeRow(1, 1)
		expect(props.qset.dimensions.x).toEqual(1)

	})

	test('CreatorTable renders 1x1 table, unable to remove 1st column', () => {
		const qset = makeQset()

		const props = {
			qset,
			cellData: jest.fn(),
		}
		const component = mount(<Table {... props}/>)

		expect(props.qset.dimensions.y).toEqual(1)
		component.instance().removeColumn(1, 1)
		expect(props.qset.dimensions.y).toEqual(1)

	})

	test('CreatorTable renders 1x1 table to append a row', () => {
		const qset = makeQset()

		const props = {
			qset,
			cellData: jest.fn(),
		}
		const component = mount(<Table {... props}/>)

		expect(props.qset.dimensions.x).toEqual(1)
		component.instance().appendRow()
		expect(props.qset.dimensions.x).toEqual(2)

	})

	test('CreatorTable renders 1x1 table to append a column', () => {
		const qset = makeQset()

		const props = {
			qset,
			cellData: jest.fn(),
		}
		const component = mount(<Table {... props}/>)

		expect(props.qset.dimensions.y).toEqual(1)
		component.instance().appendColumn()
		expect(props.qset.dimensions.y).toEqual(2)

	})

	test('CreatorTable renders 2x1 table to remove a row when focus is on another row', () => {
		const qset = makeQset()

		const props = {
			qset,
			cellData: jest.fn(),
		}
		const component = mount(<Table {... props}/>)

		component.instance().appendRow()
		expect(props.qset.dimensions.x).toEqual(2)
		component.instance().removeRow(1, 0)
		expect(props.qset.dimensions.x).toEqual(1)

	})

	test('CreatorTable renders 2x1 table to remove a row when focus is on the same row', () => {
		const qset = makeQset()

		const props = {
			qset,
			cellData: jest.fn(),
		}
		const component = mount(<Table {... props}/>)

		component.instance().appendRow()
		expect(props.qset.dimensions.x).toEqual(2)
		component.instance().removeRow(2, 0)
		expect(props.qset.dimensions.x).toEqual(1)

	})

	test('CreatorTable renders 1x2 table to remove a column when focus is on another row', () => {
		const qset = makeQset()

		const props = {
			qset,
			cellData: jest.fn(),
		}
		const component = mount(<Table {... props}/>)

		component.instance().appendColumn()
		expect(props.qset.dimensions.y).toEqual(2)
		component.instance().removeColumn(0, 1)
		expect(props.qset.dimensions.y).toEqual(1)

	})

	test('CreatorTable renders 1x2 table to remove a column when focus is on the same column', () => {
		const qset = makeQset()

		const props = {
			qset,
			cellData: jest.fn(),
		}
		const component = mount(<Table {... props}/>)

		component.instance().appendColumn()
		expect(props.qset.dimensions.y).toEqual(2)
		component.instance().removeColumn(0, 2)
		expect(props.qset.dimensions.y).toEqual(1)

	})

	test('CreatorTable focuses on non-existant cell', () => {
		const qset = makeQset()

		const props = {
			qset,
			cellData: jest.fn(),
		}
		const component = mount(<Table {... props}/>)

		const retVal = component.instance().focusOnCell(-1, 0)
		expect(retVal).toEqual(0)

	})

})
