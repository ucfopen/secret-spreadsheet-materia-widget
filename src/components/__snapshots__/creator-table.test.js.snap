// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatorTable component CreatorTable renders 1x1 table, center aligned, no header 1`] = `
<div
  className="table"
>
  <form>
    <div
      className="table-elements"
    >
      <table
        onBlur={[Function]}
      >
        <tbody>
          <tr>
            <td
              className="center-align  "
            >
              <div
                className="cell"
                onKeyDown={[Function]}
              >
                <input
                  className="row-0 col-0"
                  onChange={[Function]}
                  placeholder="A1"
                  type="text"
                  value=""
                />
                <div
                  className="checkbox"
                  onClick={[Function]}
                >
                  <input
                    checked={false}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    type="checkbox"
                  />
                  Hide
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
      <button
        hidden={true}
      />
      <div
        className="column-buttons"
      >
        <button
          onClick={[Function]}
        >
          <svg
            height="28"
            id="svg"
            version="1.1"
            viewBox="0 0 400 400"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <g
              id="svgg"
            >
              <path
                d="M85.333 85.333 C 19.189 151.478,19.189 248.522,85.333 314.667 C 190.592 419.925,360.000 349.202,360.000 200.000 C 360.000 50.798,190.592 -19.925,85.333 85.333 M213.333 160.000 C 213.333 177.778,222.222 186.667,240.000 186.667 C 254.667 186.667,266.667 192.667,266.667 200.000 C 266.667 207.333,254.667 213.333,240.000 213.333 C 222.222 213.333,213.333 222.222,213.333 240.000 C 213.333 254.667,207.333 266.667,200.000 266.667 C 192.667 266.667,186.667 254.667,186.667 240.000 C 186.667 222.222,177.778 213.333,160.000 213.333 C 145.333 213.333,133.333 207.333,133.333 200.000 C 133.333 192.667,145.333 186.667,160.000 186.667 C 177.778 186.667,186.667 177.778,186.667 160.000 C 186.667 145.333,192.667 133.333,200.000 133.333 C 207.333 133.333,213.333 145.333,213.333 160.000 "
                id="path0"
              />
            </g>
          </svg>
        </button>
        <button
          onClick={[Function]}
        >
          <svg
            height="28"
            id="svg"
            version="1.1"
            viewBox="0, 0, 400,400"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <g
              id="svgg"
            >
              <path
                d="M85.333 85.333 C 19.189 151.478,19.189 248.522,85.333 314.667 C 190.592 419.925,360.000 349.202,360.000 200.000 C 360.000 50.798,190.592 -19.925,85.333 85.333 M280.000 200.000 C 280.000 207.407,244.444 213.333,200.000 213.333 C 155.556 213.333,120.000 207.407,120.000 200.000 C 120.000 192.593,155.556 186.667,200.000 186.667 C 244.444 186.667,280.000 192.593,280.000 200.000 "
                id="path0"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
    <div
      className="row-buttons"
    >
      <button
        onClick={[Function]}
      >
        <svg
          height="28"
          id="svg"
          version="1.1"
          viewBox="0 0 400 400"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <g
            id="svgg"
          >
            <path
              d="M85.333 85.333 C 19.189 151.478,19.189 248.522,85.333 314.667 C 190.592 419.925,360.000 349.202,360.000 200.000 C 360.000 50.798,190.592 -19.925,85.333 85.333 M213.333 160.000 C 213.333 177.778,222.222 186.667,240.000 186.667 C 254.667 186.667,266.667 192.667,266.667 200.000 C 266.667 207.333,254.667 213.333,240.000 213.333 C 222.222 213.333,213.333 222.222,213.333 240.000 C 213.333 254.667,207.333 266.667,200.000 266.667 C 192.667 266.667,186.667 254.667,186.667 240.000 C 186.667 222.222,177.778 213.333,160.000 213.333 C 145.333 213.333,133.333 207.333,133.333 200.000 C 133.333 192.667,145.333 186.667,160.000 186.667 C 177.778 186.667,186.667 177.778,186.667 160.000 C 186.667 145.333,192.667 133.333,200.000 133.333 C 207.333 133.333,213.333 145.333,213.333 160.000 "
              id="path0"
            />
          </g>
        </svg>
      </button>
      <button
        onClick={[Function]}
      >
        <svg
          height="28"
          id="svg"
          version="1.1"
          viewBox="0, 0, 400,400"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <g
            id="svgg"
          >
            <path
              d="M85.333 85.333 C 19.189 151.478,19.189 248.522,85.333 314.667 C 190.592 419.925,360.000 349.202,360.000 200.000 C 360.000 50.798,190.592 -19.925,85.333 85.333 M280.000 200.000 C 280.000 207.407,244.444 213.333,200.000 213.333 C 155.556 213.333,120.000 207.407,120.000 200.000 C 120.000 192.593,155.556 186.667,200.000 186.667 C 244.444 186.667,280.000 192.593,280.000 200.000 "
              id="path0"
            />
          </g>
        </svg>
      </button>
    </div>
    <input
      hidden={true}
      type="submit"
    />
  </form>
</div>
`;

exports[`CreatorTable component CreatorTable renders 1x1 table, left aligned, header 1`] = `
<div
  className="table"
>
  <form>
    <div
      className="table-elements"
    >
      <table
        onBlur={[Function]}
      >
        <tbody>
          <tr>
            <td
              className=" header-cell "
            >
              <div
                className="cell"
                onKeyDown={[Function]}
              >
                <input
                  className="row-0 col-0"
                  onChange={[Function]}
                  placeholder="A1"
                  type="text"
                  value=""
                />
                <div
                  className="checkbox"
                  onClick={[Function]}
                >
                  <input
                    checked={false}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    type="checkbox"
                  />
                  Hide
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
      <button
        hidden={true}
      />
      <div
        className="column-buttons"
      >
        <button
          onClick={[Function]}
        >
          <svg
            height="28"
            id="svg"
            version="1.1"
            viewBox="0 0 400 400"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <g
              id="svgg"
            >
              <path
                d="M85.333 85.333 C 19.189 151.478,19.189 248.522,85.333 314.667 C 190.592 419.925,360.000 349.202,360.000 200.000 C 360.000 50.798,190.592 -19.925,85.333 85.333 M213.333 160.000 C 213.333 177.778,222.222 186.667,240.000 186.667 C 254.667 186.667,266.667 192.667,266.667 200.000 C 266.667 207.333,254.667 213.333,240.000 213.333 C 222.222 213.333,213.333 222.222,213.333 240.000 C 213.333 254.667,207.333 266.667,200.000 266.667 C 192.667 266.667,186.667 254.667,186.667 240.000 C 186.667 222.222,177.778 213.333,160.000 213.333 C 145.333 213.333,133.333 207.333,133.333 200.000 C 133.333 192.667,145.333 186.667,160.000 186.667 C 177.778 186.667,186.667 177.778,186.667 160.000 C 186.667 145.333,192.667 133.333,200.000 133.333 C 207.333 133.333,213.333 145.333,213.333 160.000 "
                id="path0"
              />
            </g>
          </svg>
        </button>
        <button
          onClick={[Function]}
        >
          <svg
            height="28"
            id="svg"
            version="1.1"
            viewBox="0, 0, 400,400"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <g
              id="svgg"
            >
              <path
                d="M85.333 85.333 C 19.189 151.478,19.189 248.522,85.333 314.667 C 190.592 419.925,360.000 349.202,360.000 200.000 C 360.000 50.798,190.592 -19.925,85.333 85.333 M280.000 200.000 C 280.000 207.407,244.444 213.333,200.000 213.333 C 155.556 213.333,120.000 207.407,120.000 200.000 C 120.000 192.593,155.556 186.667,200.000 186.667 C 244.444 186.667,280.000 192.593,280.000 200.000 "
                id="path0"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
    <div
      className="row-buttons"
    >
      <button
        onClick={[Function]}
      >
        <svg
          height="28"
          id="svg"
          version="1.1"
          viewBox="0 0 400 400"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <g
            id="svgg"
          >
            <path
              d="M85.333 85.333 C 19.189 151.478,19.189 248.522,85.333 314.667 C 190.592 419.925,360.000 349.202,360.000 200.000 C 360.000 50.798,190.592 -19.925,85.333 85.333 M213.333 160.000 C 213.333 177.778,222.222 186.667,240.000 186.667 C 254.667 186.667,266.667 192.667,266.667 200.000 C 266.667 207.333,254.667 213.333,240.000 213.333 C 222.222 213.333,213.333 222.222,213.333 240.000 C 213.333 254.667,207.333 266.667,200.000 266.667 C 192.667 266.667,186.667 254.667,186.667 240.000 C 186.667 222.222,177.778 213.333,160.000 213.333 C 145.333 213.333,133.333 207.333,133.333 200.000 C 133.333 192.667,145.333 186.667,160.000 186.667 C 177.778 186.667,186.667 177.778,186.667 160.000 C 186.667 145.333,192.667 133.333,200.000 133.333 C 207.333 133.333,213.333 145.333,213.333 160.000 "
              id="path0"
            />
          </g>
        </svg>
      </button>
      <button
        onClick={[Function]}
      >
        <svg
          height="28"
          id="svg"
          version="1.1"
          viewBox="0, 0, 400,400"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <g
            id="svgg"
          >
            <path
              d="M85.333 85.333 C 19.189 151.478,19.189 248.522,85.333 314.667 C 190.592 419.925,360.000 349.202,360.000 200.000 C 360.000 50.798,190.592 -19.925,85.333 85.333 M280.000 200.000 C 280.000 207.407,244.444 213.333,200.000 213.333 C 155.556 213.333,120.000 207.407,120.000 200.000 C 120.000 192.593,155.556 186.667,200.000 186.667 C 244.444 186.667,280.000 192.593,280.000 200.000 "
              id="path0"
            />
          </g>
        </svg>
      </button>
    </div>
    <input
      hidden={true}
      type="submit"
    />
  </form>
</div>
`;
